fluentd/lib/fluent-plugin-collected/lib/fluent/plugin/in_collected_tail_monitor.rb

```diff
@@ -0,0 +1,116 @@
+require 'fluent/plugin/input'
+require 'fluent/plugin/in_monitor_agent'
+require 'fluent/plugin/prometheus'
+
+module Fluent::Plugin
+  # Collect log_collected_bytes_total metric from in_tail plugin.
+  class CollectedTailMonitorInput < Fluent::Plugin::Input
+    Fluent::Plugin.register_input('collected_tail_monitor', self)
+    include Fluent::Plugin::PrometheusLabelParser
+
+    helpers :timer
+
+    #ideally interval must be rotatewait > 0 ? rotatewait/2 : 1.0 and if interval >= rotate_wait we should log a warning. For rotatewait = 5 sec, interval can be set to 2 sec
+    config_param :interval, :time, default: 2
+    attr_reader :registry
+
+    MONITOR_IVARS = [
+      :tails,
+    ]
+
+    def initialize
+      super
+      @registry = ::Prometheus::Client.registry
+    end
+
+    def multi_workers_ready?
+      true
+    end
+
+    def configure(conf)
+      super
+      hostname = Socket.gethostname
+      expander_builder = Fluent::Plugin::Prometheus.placeholder_expander(log)
+      expander = expander_builder.build({ 'hostname' => hostname, 'worker_id' => fluentd_worker_id })
+      @base_labels = parse_labels_elements(conf)
```

Comment:
It appears we are missing a configuration that is desired here and possibly declared as [1].  What is the intent of 'base_labels'?  How do we apply it in practice?

[1] https://docs.fluentd.org/plugin-development/api-config-types#hash

Reply:
base_labels merge method allows us to add label to collected_bytes_total metric time series. I have added namespace, podname, containername as additional labels same as what we have done for log-file-metric-exporter published logged_bytes_total metric.

Reply:
Reviewing the source of why this is needed, this allows for static configuration of a set of labels to be included with the metrics [1].  No need to provide additional conf declaration

[1] https://github.com/openshift/origin-aggregated-logging/tree/master/fluentd/vendored_gem_src/fluent-plugin-prometheus#labels